controllers/datafoundation.go

```diff
@@ -171,6 +183,9 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 
 	} else {
+		if err := r.reconcileWatchers(); err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
The code in this func needs to run before anything else in the reconcilePhases func as it is an assumption that we need to ensure before having any reconcile code.